cargo-screeps/src/config.rs

```diff
@@ -0,0 +1,159 @@
+use failure::{self, ResultExt};
+
+use toml;
+
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct BuildConfiguration {
+    pub output_wasm_file: PathBuf,
+    pub output_js_file: PathBuf,
+}
+
+impl Default for BuildConfiguration {
+    fn default() -> Self {
+        BuildConfiguration {
+            output_wasm_file: "compiled.wasm".into(),
+            output_js_file: "main.js".into(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+struct FileUploadConfiguration {
+    username: String,
+    password: String,
+    branch: String,
+    #[serde(default = "default_hostname")]
+    hostname: String,
+    #[serde(default)]
+    ssl: Option<bool>,
+    port: Option<i32>,
+    #[serde(default = "default_ptr")]
+    ptr: bool,
+}
+
+fn default_hostname() -> String {
+    "screeps.com".to_owned()
+}
+
+fn default_ptr() -> bool {
+    false
+}
+
+#[derive(Clone, Debug)]
+pub struct UploadConfiguration {
+    pub username: String,
+    pub password: String,
+    pub hostname: String,
+    pub branch: String,
+    pub ssl: bool,
+    pub port: i32,
+    pub ptr: bool,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct CopyConfiguration {
+    pub destination: PathBuf,
+    pub branch: String,
```

Comment:
Unless I'm mistaken, the `branch` isn't used (at least, not in `copy.rs`)

Reply:
I think it should be, at least. It's used as a sub-directory under destination to stick files in.

Reply:
Indeed, I had missed that. Nevermind!